### ISSUE TITLE
Average Loss explanation with one pass. One against all, OAA.

### ISSUE THREAD

**Issue by japerez20** [JohnLangford/vowpal_wabbit#775]
Hello :
I  am using -oaa with 5 states in order to build a classification model.
I have read all the tutorials related to vw and I have some questions related to average loss.

I am aware the model is built online and  several other features, some  of the features explained in Arielf presentation (ppt) were:
  
    A.- Error estimate is realistic (test like)
    B.- No need to train vs test or K-fold cross validate.

The questions below  are related with the option one pass.
What would be the interpretation of the average loss.
What would be holdout data (1 every 10 observations by default?).

Is the average loss related to some performance metric in the holdout data ?
By perfomance metric I mean, F1 score, accuracy, precision, recall.
How can I associate the average loss in one pass with the quality of the fit.

How can I connect the average loss with points A and B above?

Thank you,
ja

**Comment by martinpopel** [comment]
I am not sure what is the (main) question or problem. (GitHub issues should be for concrete problems or feature suggestions, note that there is a [VW mailing list](https://groups.yahoo.com/neo/groups/vowpal_wabbit/info) and [vowpalwabbit tag at stackoverflow](http://stackoverflow.com/questions/tagged/vowpalwabbit).)

If you use one pass and no `--holdout_off` option, VW computes 'progressive validation loss', which is a realistic estimate of a holdout loss (computed on a separate test set). If you need exact holdout/test loss on a given test set then simply train a model and then use `vw -d my_testset -i my_model -t`.

The loss reported by VW can be a squared loss, logistic loss, hinge, quantile loss or 0-1 loss depending on the option `--loss_function` you used and possibly also `--oaa`, which switches the reporting to 0-1 loss. For reporting F1, precision, recall or AUC, I suggest [using perf](http://stackoverflow.com/questions/24264516/calculating-auc-when-using-vowpal-wabbit/24660520#24660520).

**Comment by JohnLangford** [comment]
Closing this for now: no clear issue.

**Comment by arielf** [comment]
Just to help the OP a bit more:

If you run with `--progress 1` and extract the two columns (5, and 6) titled:  `current label` and `current predict`, you can feed them to `perf` as Martin suggested above.  Be aware that your initial loss may be high since `vw` has not learned anything yet.  So you may want to caclulate  F1, precision, or AUC , only after some convergence has taken place, again as Martin suggested.
